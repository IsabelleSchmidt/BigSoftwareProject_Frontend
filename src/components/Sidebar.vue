<template>
  <div class="sidebar">
    <h4>Produkte</h4>
    <ul>
      <li>
        <router-link
          id="link"
          :to="{
            path: '/product',
            query: { room: 'all', producttype: 'Pflanze', name: 'none' },
          }"
          >Pflanzen</router-link
        >
      </li>
      <li>
        <router-link
          id="link"
          :to="{
            path: '/product',
            query: { room: 'all', producttype: 'Tisch', name: 'none' },
          }"
          >Tische</router-link
        >
      </li>
      <li>
        <router-link
          id="link"
          :to="{
            path: '/product',
            query: { room: 'all', producttype: 'Stuhl', name: 'none' },
          }"
          >St체hle</router-link
        >
      </li>
      <li>
        <router-link
          id="link"
          :to="{
            path: '/product',
            query: { room: 'all', producttype: 'Bett', name: 'none' },
          }"
          >Betten</router-link
        >
      </li>
      <li>
        <router-link
          id="link"
          :to="{
            path: '/product',
            query: { room: 'all', producttype: 'Dekoration', name: 'none' },
          }"
          >Deko</router-link
        >
      </li>
      <li>
        <router-link
          id="link"
          :to="{
            path: '/product',
            query: { room: 'all', producttype: 'Sofa/Couch', name: 'none' },
          }"
          >Sofa/Couch</router-link
        >
      </li>
      <li>
        <router-link
          id="link"
          :to="{
            path: '/product',
            query: {
              room: 'all',
              producttype: 'Schrank/Kommode',
              name: 'none',
            },
          }"
          >Schr채nke/Kommoden</router-link
        >
      </li>
    </ul>

    <br />

    <h4>R채ume</h4>
    <ul>
      <li>
        <router-link
          id="link"
          :to="{
            path: '/product',
            query: { room: 'Bad', producttype: 'all', name: 'none' },
          }"
          >Bad</router-link
        >
      </li>
      <li>
        <router-link
          id="link"
          :to="{
            path: '/product',
            query: { room: 'Schlafzimmer', producttype: 'all', name: 'none' },
          }"
          >Schlafzimmer</router-link
        >
      </li>
      <li>
        <router-link
          id="link"
          :to="{
            path: '/product',
            query: { room: 'Kueche', producttype: 'all', name: 'none' },
          }"
          >K체che</router-link
        >
      </li>
      <li>
        <router-link
          id="link"
          :to="{
            path: '/product',
            query: { room: 'Wohnzimmer', producttype: 'all', name: 'none' },
          }"
          >Wohnzimmer</router-link
        >
      </li>
      <li>
        <router-link
          id="link"
          :to="{
            path: '/product',
            query: { room: 'Esszimmer', producttype: 'all', name: 'none' },
          }"
          >Esszimmer</router-link
        >
      </li>
      <li>
        <router-link
          id="link"
          :to="{
            path: '/product',
            query: { room: 'Arbeitszimmer', producttype: 'all', name: 'none' },
          }"
          >Arbeitszimmer</router-link
        >
      </li>
    </ul>
  </div>
</template>

<style scoped lang="scss">
.sidebar {
  text-align: left;
  color: black;
  float: left;
  background-color: #f3f3f3;
  padding: 1.5%;
  max-width: 15%;
  margin-left: 20px;
  margin-right: 30px;
  margin-top: 15px;
  border-radius: 5px;
  position: fixed;
}
ul {
  list-style-type: none;
  margin: 0px;
}
#link {
  color: black;
  text-decoration: none;
  list-style: none;
  list-style-type: none;
}
#link:hover {
  color: #6f492d;
  text-decoration: none;
}
li {
  font-size: 1.25vw;
}
h4 {
  font-size: 1.5vw;
  margin: 10px;
}
</style>