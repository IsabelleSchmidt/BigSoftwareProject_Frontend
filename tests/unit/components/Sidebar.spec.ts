import { shallowMount } from "@vue/test-utils"

import Sidebar from "../../../src/components/Sidebar.vue"

describe('Sidebar.vue Test', () => {

    const cmp = shallowMount(Sidebar)

    test('If the HTML structur is right', () => {
        expect(cmp.html()).toMatchSnapshot()
    })

    test('If HTML contains product and rooms', () => {
        expect(cmp.html()).toContain('Produkte')
        expect(cmp.html()).toContain('RÃ¤ume')
    })
})
