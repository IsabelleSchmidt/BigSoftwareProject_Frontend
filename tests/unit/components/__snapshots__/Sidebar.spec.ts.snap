// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar.vue Test If the HTML structur is right 1`] = `
<div class="sidebar">
  <h4>Produkte</h4>
  <ul>
    <li>
      <router-link id="link" to="[object Object]">Pflanzen</router-link>
    </li>
    <li>
      <router-link id="link" to="[object Object]">Tische</router-link>
    </li>
    <li>
      <router-link id="link" to="[object Object]">St체hle</router-link>
    </li>
    <li>
      <router-link id="link" to="[object Object]">Betten</router-link>
    </li>
    <li>
      <router-link id="link" to="[object Object]">Deko</router-link>
    </li>
    <li>
      <router-link id="link" to="[object Object]">Sofa/Couch</router-link>
    </li>
    <li>
      <router-link id="link" to="[object Object]">Schr채nke/Kommoden</router-link>
    </li>
  </ul><br>
  <h4>R채ume</h4>
  <ul>
    <li>
      <router-link id="link" to="[object Object]">Bad</router-link>
    </li>
    <li>
      <router-link id="link" to="[object Object]">Schlafzimmer</router-link>
    </li>
    <li>
      <router-link id="link" to="[object Object]">K체che</router-link>
    </li>
    <li>
      <router-link id="link" to="[object Object]">Wohnzimmer</router-link>
    </li>
    <li>
      <router-link id="link" to="[object Object]">Esszimmer</router-link>
    </li>
    <li>
      <router-link id="link" to="[object Object]">Arbeitszimmer</router-link>
    </li>
  </ul>
</div>
`;
